### PR TITLE
🐛 Fix remediation text when Scorecard is run multiple times within a program

### LINKED ISSUE #2152: Using Scorecard GitHub Action reduces the Token-Permissions score
**Describe the bug**

urllib3 recently added the [Scorecard GitHub Action](https://github.com/urllib3/urllib3/blob/main/.github/workflows/scorecards.yml) to its repository. After adding this we observed our "Token-Permissions" score being 9/10 instead of the previous value of 10/10. This was due to:

> Warn: jobLevel 'security-events' permission set to 'write': .github/workflows/scorecards.yml:16: update your workflow using https://app.stepsecurity.io/secureworkflow/syndicate-lang/syndicate-js/scorecards.yml/main?enable=permissions

This is because the action wants the permission `security-events: write` which is a permission that is docked points if given to any GitHub Action.

**Reproduction steps**
1. Add the Scorecard GitHub Action w/ `security-events: write`
2. Run Scorecard on the repository
3. Observe an imperfect score due to Scorecard GitHub Action config

**Expected behavior**

Using the Scorecard GitHub Action shouldn't reduce the project's scorecard score.

### PR DESCRIPTION
**Pull request by spencerschrock**
#### What kind of change does this PR introduce?
bug fix for incorrect repo info in remediation text. the problem affected the weekly scorecard cron job

- [X] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?
The remediation text used to reflect the first repo / branch  processed by a scorecard cron worker.

#### What is the new behavior (if this is a feature change)?**
The remediation text will now match the repo/branch being analyzed. 

- [X] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes
#2152 

#### Special notes for your reviewer
This is a temporary fix, and modifies the function signatures for `evaluation.PinningDependencies` and `evaluation.TokenPermissions`. I don't think anyone besides ourselves are using these exported functions.

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
NONE
```

### CODE REVIEW

**Comment by azeemshaikh38** [comment]
Could we simply extend [`PinnedDepsRawData`](https://github.com/ossf/scorecard/blob/main/checker/raw_result.go#L92) and [`TokenPermissionsRawData`](https://github.com/ossf/scorecard/blob/main/checker/raw_result.go#L252) to include `RemediationMetadata` struct?

This isn't ideal, and really the raw data struct should have a common struct to hold this info. But atleast this avoids passing in the CheckRequest to `evaluation`.

File: remediation/remediations.go [hunk 1]
```diff
@@ -42,53 +33,51 @@ var (
...
-		uri := c.RepoClient.URI()
-		parts := strings.Split(uri, "/")
-		if len(parts) != 3 {
-			setupErr = fmt.Errorf("%w: empty: %s", errInvalidArg, uri)
-			return
-		}
-		repo = fmt.Sprintf("%s/%s", parts[1], parts[2])
-	})
-	return setupErr
+func Setup(c *checker.CheckRequest) (RemediationMetadata, error) {
```
Comment (azeemshaikh38) [inline remediation/remediations.go:43]: Nit: change name to `New` ?

**Review by azeemshaikh38 — APPROVED** [review]

**Comment by azeemshaikh38** [comment]
> > Could we simply extend [`PinnedDepsRawData`](https://github.com/ossf/scorecard/blob/main/checker/raw_result.go#L92) and [`TokenPermissionsRawData`](https://github.com/ossf/scorecard/blob/main/checker/raw_result.go#L252) to include `RemediationMetadata` struct?
> > This isn't ideal, and really the raw data struct should have a common struct to hold this info. But atleast this avoids passing in the CheckRequest to `evaluation`.
> 
> I was just following your quick fix comments in #2152. Also, I was hesitant to touch RawData for fear of messing up the json/cron job flow.
> 
> > But I know it's not an easy change and might take time. So, let's remove the init() fn and all the global vars for now (that should solve the data race issue too). We need to fix this before EOW so that the cron job picks up the fix.
> > Once the fix is in, let's reconsider how to solve it the right way using RawData. Wdyt?

SGTM, approved. Could we have a follow-up PR to address this?

**Comment by spencerschrock** [comment]
> SGTM, approved. Could we have a follow-up PR to address this?

👍

File: checks/evaluation/pinned_dependencies.go [hunk 2]
```diff
@@ -133,10 +136,10 @@ func PinningDependencies(name string, dl checker.DetailLogger,
 		"dependency not pinned by hash detected", score, checker.MaxResultScore)
 }
 
-func generateRemediation(rr *checker.Dependency) *checker.Remediation {
+func generateRemediation(remediaitonMd remediation.RemediationMetadata, rr *checker.Dependency) *checker.Remediation {
```
Comment (laurentsimon) [inline checks/evaluation/pinned_dependencies.go:139]: nit: remediaitonMd

**Pull request merged**